### PR TITLE
Fix Android runtime errors

### PR DESCRIPTION
**Pull request by nickmeinhold**
* Update & remove redundant dependencies
- removed unused firebase_messaging package to hopefully avoid
version conflicts
- updated gradle, compileSDKVersion, targetSDKVersion

### CODE REVIEW

File: lib/utils/wrappers/firebase_wrapper.dart [hunk 1]
```diff
@@ -1,7 +1,15 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:firebase_core/firebase_core.dart';
 
 /// Wraps the [Firebase] static function so we can optionally inject a mock in
 /// the [CrowdLeagueApp] widget.
```
Comment (T-Davis) [inline lib/utils/wrappers/firebase_wrapper.dart:5]: Small nit for CrowdLeagueApp

File: lib/utils/wrappers/firebase_wrapper.dart [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
 class FirebaseWrapper {
-  Future<FirebaseApp> init() => Firebase.initializeApp();
+  Future<FirebaseApp> init() async {
+    final app = await Firebase.initializeApp();
+
+    // TODO: remove workaround
```
Comment (T-Davis) [inline lib/utils/wrappers/firebase_wrapper.dart:10]: Apologies if this was already explained, but could you tell me what this workaround is for?

**Review by T-Davis — CHANGES_REQUESTED** [review]
Thanks for the PR!
I left a couple of comments to address

File: lib/utils/wrappers/firebase_wrapper.dart (hunk 1 above)
Comment (nickmeinhold) [reply lib/utils/wrappers/firebase_wrapper.dart:5]: Thanks mate!  Do you think we should agree to try and make these nits as "suggested changes", probably ends up being less work right, given that you're already at the code to make the suggestion?   I'll do it now, just to be clear, just thinking for  the future

Comment (nickmeinhold) [inline lib/utils/wrappers/firebase_wrapper.dart:5]:
```suggestion
/// the [AppWidget].
```

File: lib/utils/wrappers/firebase_wrapper.dart (hunk 2 above)
Comment (nickmeinhold) [reply lib/utils/wrappers/firebase_wrapper.dart:10]: No worries!  Yeah we discussed this morning - it's the crash due to NNBD Firestore.  The url was indeed in one of the commits... one win among many losses my ailing memory :-) FirebaseExtended/flutterfire#4948

Comment (nickmeinhold) [reply lib/utils/wrappers/firebase_wrapper.dart:10]: I went ahead and committed the change, not sure if that was protocol... happy to be corrected!

**Review by nickmeinhold — COMMENTED** [review]
I think it's good when you're ready mate!

File: lib/utils/wrappers/firebase_wrapper.dart (hunk 1 above)
Comment (T-Davis) [reply lib/utils/wrappers/firebase_wrapper.dart]: Yep, definitely! I should've

File: lib/utils/wrappers/firebase_wrapper.dart (hunk 2 above)
Comment (T-Davis) [reply lib/utils/wrappers/firebase_wrapper.dart:10]: All good! Better to have a working app I reckon 😉

**Review by T-Davis — APPROVED** [review]
Looks great! Thanks for getting us back up and running 👍

**Pull request merged**